### ISSUE TITLE
Can aggregate function whitelist support configurable?

### ISSUE THREAD

**Issue by ChenHuajun** [citusdata/citus#3916]
Now, aggregation functions that support parallel execution in Citus are configured through a hard-coded whitelist in `multi_logical_optimizer.h`.

We need to use the aggregate function in `pg_roaringbitmap`

https://github.com/ChenHuajun/pg_roaringbitmap
-`rb_and_agg`
-`rb_or_agg`

Could this simple whitelist of aggregation functions satisfying the associativity be made configurable?

**Comment by metdos** [comment]
@ChenHuajun I suppose you don't add distribution column to your queries?
https://docs.citusdata.com/en/v9.3/develop/reference_sql.html?highlight=coordinator_aggregation_strategy#aggregate-functions

**Comment by onurctirtir** [comment]
Citus currently does not support parallel aggregate execution on worker nodes for aggregates having their `state_data_type` to be `internal`

**Comment by ChenHuajun** [comment]
> @ChenHuajun I suppose you don't add distribution column to your queries?
> https://docs.citusdata.com/en/v9.3/develop/reference_sql.html?highlight=coordinator_aggregation_strategy#aggregate-functions

Yes.  As @onurctirtir said what i need is support of aggregates with stype = internal

**Comment by marcocitus** [comment]
More context in https://github.com/citusdata/citus/issues/120

We should add a metadata table that contains "safe" aggregates (and check it in AggregateEnabledCustom) and then implement serialization in worker_partial_agg_ffunc and deserialization in coord_combine_agg_sfunc.
